### PR TITLE
fix: replaces internal focus lock hook with react-focus-on

### PR DESCRIPTION
**Pull request by dzucconi**
We already use the scroll isolation part of this library so I've just moved us over to use all of it's capabilities including focus trapping; which was breaking in subtle ways.

### Breaking Changes

* Removes `useFocusLock`

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @artsy/palette-charts@29.0.0-canary.1269.26463.0
  npm install @artsy/palette@30.0.0-canary.1269.26463.0
  # or 
  yarn add @artsy/palette-charts@29.0.0-canary.1269.26463.0
  yarn add @artsy/palette@30.0.0-canary.1269.26463.0
  ```
</details>

### CODE REVIEW

File: packages/palette/src/elements/Modal/__tests__/ModalBase.test.tsx [hunk 1]
```diff
@@ -1,11 +1,18 @@
 import { mount } from "enzyme"
 import React, { useState } from "react"
-import { act } from "react-dom/test-utils"
 import { ModalBase } from "../ModalBase"
 
+<<<<<<< Updated upstream
```
Comment (dzucconi) [inline packages/palette/src/elements/Modal/__tests__/ModalBase.test.tsx:5]: Woops

**Comment by dzucconi** [comment]
OK — This is passing in Integrity locally.

**Comment by artsyit** [comment]
:rocket: PR was released in [`@artsy/palette-charts@29.0.0`](https://github.com/artsy/palette/releases/tag/%40artsy/palette-charts%4029.0.0), [`@artsy/palette@30.0.0`](https://github.com/artsy/palette/releases/tag/%40artsy/palette%4030.0.0) :rocket:

**Comment by damassi** [comment]
real nice

**Pull request merged**